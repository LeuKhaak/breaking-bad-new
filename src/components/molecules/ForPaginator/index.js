import React from "react";
import * as styles from "./styles.module.scss";
import Btn from "src/components/atoms/Btn";

function ForPaginator({ perPage, amounts, changeLimitPages }) {
  return (
    <div className={styles.amount}>
      <span className={styles.show}>Show cards:</span>
      {amounts &&
        amounts.map((el) => (
          <div key={el} className={styles.pages}>
            <Btn
              btnText={el}
              btnStyle={el === perPage ? "pageNumberChecked" : "pageNumber"}
              onClick={changeLimitPages}
              arg={el}
            />
          </div>
        ))}
    </div>
  );
}

export default ForPaginator;
