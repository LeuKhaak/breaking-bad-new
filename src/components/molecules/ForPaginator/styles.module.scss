@import "../../../assets/styles/styles.scss";
.amount {
  display: flex;
  align-items: center;
  @include mobile {
    margin-bottom: 1.5rem;
  }
  .show {
    display: block;
    margin-right: 1rem;
  }
  .pages {
    display: flex;
    align-items: center;
  }
}
