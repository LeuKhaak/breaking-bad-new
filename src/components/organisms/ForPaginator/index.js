import React from "react";
import { useSelector, useDispatch } from "react-redux";
import * as styles from "./styles.module.scss";
import Btn from "src/components/atoms/Btn";

function ForPaginator() {
  const perPage = useSelector((state) => state.paginator.perPage);
  const amounts = [5, 10, 15, 20];
  return (
    <div className={styles.amount}>
      <span className={styles.show}>Show cards:</span>
      {amounts &&
        amounts.map((el) => (
          <div key={el} className={styles.pages}>
            <Btn
              btnText={el}
              btnStyle={el === perPage ? "pageNumberChecked" : "pageNumber"}
            />
          </div>
        ))}
    </div>
  );
}

export default ForPaginator;
