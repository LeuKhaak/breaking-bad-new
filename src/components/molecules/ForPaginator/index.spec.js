import React from "react";
import { shallow } from "enzyme";
import ForPaginator from ".";

let component = null;
const setComponent = (props) => shallow(<ForPaginator {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("ForPaginator testing:", () => {
  test("is ForPaginator contain items", () => {
    component = setComponent({
      amounts: [5, 10, 15, 20],
      perPage: 10,
    });
    expect(component).toMatchSnapshot();
  });
});
