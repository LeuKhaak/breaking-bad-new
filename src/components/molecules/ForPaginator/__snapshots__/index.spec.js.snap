// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForPaginator testing: is ForPaginator contain items 1`] = `
<div>
  <span>
    Show cards:
  </span>
  <div
    id="b5"
    key="5"
  >
    <Btn
      arg={5}
      btnStyle="pageNumber"
      btnText={5}
    />
  </div>
  <div
    id="b10"
    key="10"
  >
    <Btn
      arg={10}
      btnStyle="pageNumberChecked"
      btnText={10}
    />
  </div>
  <div
    id="b15"
    key="15"
  >
    <Btn
      arg={15}
      btnStyle="pageNumber"
      btnText={15}
    />
  </div>
  <div
    id="b20"
    key="20"
  >
    <Btn
      arg={20}
      btnStyle="pageNumber"
      btnText={20}
    />
  </div>
</div>
`;
